### ISSUE TITLE
systemd_unit property seems conflicting with "real" systemd_unit resource in 7.0.0

### ISSUE THREAD

**Issue by BarthV** [sous-chefs/haproxy#378]
### Cookbook version
7.0.0

### Chef-client version
12.18.31

### Platform Details
Bare-metal , CentOS 7.4 , using systemd

### Scenario:
Upgrading from haproxy `6.4.0` to `7.0.0`

### Steps to Reproduce:
Use a `haproxy_service` resource as in documentation 

### Expected Result:
haproxy reloading & running 

### Actual Result:
```
  * haproxy_service[haproxy] action createINFO: Processing haproxy_service[haproxy] action create (policy-foobar::my-recipe line 36)

    
    ================================================================================
    Error executing action `create` on resource 'haproxy_service[haproxy]'
    ================================================================================
    
    RuntimeError
    ------------
    Property `systemd_unit` of `haproxy_service[haproxy]` was incorrectly passed a block.  Possible property-resource collision.  To call a resource named `systemd_unit` either rename the property or else use `declare_resource(:systemd_unit, ...)`
    
    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/haproxy/resources/service.rb:44:in `block (2 levels) in class_from_file'
    /var/chef/cache/cookbooks/haproxy/resources/service.rb:10:in `block in class_from_file'
    
    Resource Declaration:
    ---------------------
    # In my-recipe.rb
    
     36: haproxy_service 'haproxy' do
     37:   subscribes :reload, 'template[/etc/haproxy/haproxy.cfg]', :immediate
     38: end
     39: 
    
    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/policy-foobar/recipes/my-recipe.rb:36:in `from_file'
    
    haproxy_service("haproxy") do
      action [:create]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :haproxy_service
      cookbook_name "policy-foobar"
      recipe_name "my-recipe"
      systemd_unit "[Unit]\nDescription=HAProxy Load Balancer\nDocumentation=file:/usr/share/doc/haproxy/configuration.txt.gz\nAfter=network.target syslog.service\n[Service]\nEnvironmentFile=-/etc/default/haproxy\nEnvironment=\"CONFIG=/etc/haproxy/haproxy.cfg\" \"PIDFILE=/run/haproxy.pid\"\nExecStartPre=/usr/sbin/haproxy -f $CONFIG -c -q\nExecStart=/usr/sbin/haproxy-systemd-wrapper -f $CONFIG -p $PIDFILE $OPTIONS\nExecReload=/usr/sbin/haproxy -f $CONFIG -c -q\nExecReload=/bin/kill -USR2 $MAINPID\nKillSignal=TERM\nUser=root\nWorkingDirectory=/\nKillMode=mixed\nRestart=always\n[Install]\nWantedBy=multi-user.target\n"
      config_dir "/etc/haproxy"
      config_file "/etc/haproxy/haproxy.cfg"
      bin_prefix "/usr"
      service_name "haproxy"
    end
    
    Platform:
    ---------
    x86_64-linux
```

**Comment by JohnRoesler** [comment]
@BarthV thanks for reporting this! Looks like a name collisions - probably should have thought of this before. PR coming to resolve the issue

**Comment by JohnRoesler** [comment]
Released version 7.1.0 with the fix!
